packages/react-signer/src/TxSigned.tsx

```diff
@@ -274,23 +289,26 @@ function TxSigned ({ className, currentItem, requestAddress }: Props): React.Rea
       setTimeout((): void => {
         const errorHandler = (error: Error): void => {
           console.error(error);
+
           setBusy(false);
           setError(error);
         };
 
-        try {
-          if (_unlock()) {
-            isSubmit
-              ? currentItem.payload
-                ? _onSendPayload(queueSetTxStatus, currentItem, senderInfo)
-                : _onSend(queueSetTxStatus, currentItem, senderInfo).catch(errorHandler)
-              : _onSign(queueSetTxStatus, currentItem, senderInfo).catch(errorHandler);
-          } else {
-            setBusy(false);
-          }
-        } catch (error) {
-          errorHandler(error as Error);
-        }
+        _unlock()
+          .then((isUnlocked): void => {
+            if (isUnlocked) {
+              isSubmit
+                ? currentItem.payload
+                  ? _onSendPayload(queueSetTxStatus, currentItem, senderInfo)
+                  : _onSend(queueSetTxStatus, currentItem, senderInfo).catch(errorHandler)
+                : _onSign(queueSetTxStatus, currentItem, senderInfo).catch(errorHandler);
```

Comment:
The catch here should be the one below, however I really didn't want to mess around too much with previously working.